src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -499,9 +499,26 @@ ucs_status_t uct_rc_mlx5_ep_atomic_cswap32(uct_ep_h tl_ep, uint32_t compare, uin
 
 ucs_status_t uct_rc_mlx5_ep_fence(uct_ep_h tl_ep, unsigned flags)
 {
-    uct_rc_mlx5_ep_t *ep = ucs_derived_of(tl_ep, uct_rc_mlx5_ep_t);
+    uct_rc_mlx5_ep_t *ep              = ucs_derived_of(tl_ep, uct_rc_mlx5_ep_t);
+    uct_rc_mlx5_iface_common_t *iface = ucs_derived_of(tl_ep->iface, uct_rc_mlx5_iface_common_t);
+    ucs_status_t status;
+
+    UCT_TL_EP_STAT_FENCE(ucs_derived_of(tl_ep, uct_base_ep_t));
+
+    if (iface->super.config.fence_mode == UCT_RC_FENCE_MODE_NONE) {
+        return UCS_OK;
+    }
+
+    status = uct_rc_ep_flush(&ep->super, ep->tx.wq.bb_max, flags);
```

Comment:
why need flush here?
also it is the same function as used for the verbs - can have common func on RC level

Reply:
flush is used to identify existing resources, ```uct_rc_ep_flush``` doesn't make flush - it just checks for existing ongoing ops and resources

code looks like similar to verbs fence, but used different access members... but ok, unified

Reply:
it looks a bit weird, also you increase flush counter inside the flush. If you need the same checks, better define a common func

Reply:
no problemo - added separate function :)